src/main/java/org/jabref/logic/importer/AdvancedFetcher.java

```diff
@@ -0,0 +1,38 @@
+package org.jabref.logic.importer;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.List;
+
+import org.jabref.logic.importer.fetcher.AdvancedSearchConfig;
+import org.jabref.logic.net.URLDownload;
+import org.jabref.model.entry.BibEntry;
+
+/**
+ * This interface allows SearchBasedParserFetcher fetchers to test their corresponding
+ * library APIs for their advanced search options, e.g. search in the "title" field.
+ */
+public interface AdvancedFetcher extends SearchBasedParserFetcher {
```

Comment:
As this is a special form of a `SearchBasedParserFetcher` I would suggest to also reflect it in the name: `AdvancedSearchBasedParserFetcher` (although it gets a bit long).

Moreover, I think, it's a good idea to replicate the abstract interface and introduce a `AdvancedSearchBasedFetcher` as well. Finally, these `Advanced*` classes should probably overwrite the normal `performSearch` methods and rely them to the `performAdvancedSearch` methods in order to make implementation easier, or not?

Reply:
Yeah, I totally agree with the renaming of the interface and method.

I can see your point regarding the introduction of advanced Fetcher classes, but I think that this might not work that simply, as these advanced fetchers would somehow have to parse the query, as it is a string in the standard performSearch method, into a ComplexQuery.
I think this is out of the scope of the current PR as it just tries to make the fielded search capabilities of the external library APIs testable.
But I really like your idea and I believe it will be in the scope of the thesis, just not in scope of this PR.

Reply:
Ok, we can leave it for now. What I was thinking was to simply pass the normal query as the "default field" and then call the "advanced search" method. 

Otherwise I agree, parsing complex queries should probably happen in the user interface (and should be consistent across fetchers to add even more value for users).